### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by 0bi0n3**
I'm not sure if I've done 'Part 1d - Step 6: assignment' correctly. When placing the plain-English UDTs in the provided space, the code wouldn't compile as the UDT types are not defined... e.g. "`thing 1`)". So I've left it as commented out code.

Please let me know if that is not correct!

### CODE REVIEW

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Good.  Complete the rest of the instructions in the project part

File: main.cpp [hunk 1]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+    bool isAtWar = false;
+
+    // NESTED UDT
+    struct FarmlandRevenue
+    {
+        int farmlandArces = 65;
+        int numberOfWorkers = 120;
+        float wheatSalePricePerKG = 1.8f;
+        std::string farmlandType = "Cereals and Grains";
+        bool isFarmlandInDraught = false;
```
Comment (matkatmusic) [inline main.cpp]: You do not need to include the class/struct name in any variable names.
`acres`, not `farmlandAcres`
`type`, not `farmlandType`
`isInDrought`, not `isFarmlandInDrought` 
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+    char modeSelection = 'X';
+};
+// 3 things it can do:
+//     1) adjust pitch.
+    double adjustPitch(MidiKeyboard pitchChange); // NOTE: the pitch control effects the adjustment of the functions return value.
+//     2) register key presses.
+    void isKeyPressed();
+
+//     3) register pad hit velocities.
+    float padHitAmount(MidiKeyboard padNumber); // NOTE: returns a relevant value able to influence other fuctions.
```
Comment (matkatmusic) [inline main.cpp:978]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -545,12 +430,127 @@ MOVE THEM to the space below this block comment and put them in numerical order
...
+    1) screen.
+    2) CPU.
+    3) GPU.
+    4) Memory.
+    5) USB port.
+3 things it can do:
+    1) Load a game.
+    2) Connect peripherals.
+    3) Display image.
+---------------------------------------------------------------------------------- */
```
Comment (matkatmusic) [inline main.cpp]: None of this should be here.
You were supposed to MOVE, not Copy, these plain-english descriptions
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+    char modeSelection = 'X';
+};
+// 3 things it can do:
+//     1) adjust pitch.
+    double adjustPitch(MidiKeyboard pitchChange); // NOTE: the pitch control effects the adjustment of the functions return value.
+//     2) register key presses.
+    void isKeyPressed();
 
+//     3) register pad hit velocities.
+    float padHitAmount(MidiKeyboard padNumber); // NOTE: returns a relevant value able to influence other fuctions.
```
Comment (matkatmusic) [inline main.cpp]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+        void savingSequence(int stepSize, int numberOfStepsEnabled, int numberOfLoops, int tempo);
+
+        bool savingToExternal(int stepCount, std::string fileType = "TB303.p"); // NOTE: function checks whether external device is selected for export.
+
+        void displaySavingProgress(int timeRemaining, float savingRate);
+    };
+
+SavePattern save;
+};
+// 3 things it can do:
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+//     1) program sequence of notes.
+    void programSequence(RolandTB303 sequence);
+
+//     2) apply filtering.
+    void filterSignal(float cutoffFrequency);
+
+//     3) increase || decrease playback tempo.
+    float tempoAdjust(RolandTB303 tempo); // NOTE: returns value relating to amount of temp adjustment.
```
Comment (matkatmusic) [inline main.cpp:1030]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 7]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+    std::string displayStorageAmount = " 900MBs";
+    float dataOutputPort = 76.f;
+};
+// 3 things it can do:
+//     1) export audio files.
+    void exportAudioFile(float audioData);
+//     2) record audio files.
+    void recordingOn(bool activeRecording);
+//     3) display recording space available.
+    std::string displayAvailableSpace(Dictaphone dictaphoneStorage);
```
Comment (matkatmusic) [inline main.cpp:1062]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 8]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+    int aspectRatio = 16 * 9;
+    int resolution = 1080;
+};
+// 3 things it can do:
+//     1) increase brightness
+    void brightnessUp(int currentBrightnessLevel);
+//     2) change contrast
+    void adjustContrast(Screen screenContrast);
+//     3) switch aspect ratio
+    int screenRatioSwitch(Screen screenRatio); // NOTE: return value provides back the current state of the switch.
```
Comment (matkatmusic) [inline main.cpp]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 9]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+};
+// 3 things it can do:
+//     1) Save data
+    int saveSelectedData(int selectedData); // NOTE: returns a flag to indicate save was successful.
+
+//     2) Process data
+    void interpolateData(CPU cpuDataStream);
+
+//     3) Access data
+    void openDataFile(std::string fileName);
```
Comment (matkatmusic) [inline main.cpp:1130]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 10]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+};
+// 3 things it can do:
+//     1) Increase ventilation amount
+    void fansOn(GPU gpuFans);
+
+//     2) Render graphics
+    void graphicsRendering(GPU gpu);
+
+//     3) Overlock processing speed
+    float processorOverlocking(int currentProcessorSpeed, int inputClockSpeed, GPU gpu); // NOTE: returns overlocking amount as a value.
```
Comment (matkatmusic) [inline main.cpp:1164]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 11]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+};
+// 3 things it can do:
+//     1) write data
+    int writeDataToMemory(Memory memoryDataSource); // NOTE: returns a value to signify successful writing to memory.
+
+//     2) read data
+    void openDataLocation(Memory memoryaddress);
+
+//     3) erase data
+    void deleteData(std::string selectedData);
```
Comment (matkatmusic) [inline main.cpp:1198]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 12]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+};
+// 3 things it can do:
+//     1) connect external devices
+    int connectingDevice(UsbPort usbInput); // NOTE: returns a value indicating successful or failed connection.
+
+//     2) transfer files
+    void fileTransferring(std::string fileToSend);
+
+//     3) charge devices
+    bool chargingConnectedDevice(UsbPort usbConnected); // NOTE: returns a bool value to cause system to display positive LED charging light.
```
Comment (matkatmusic) [inline main.cpp:1232]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

File: main.cpp [hunk 13]
```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
...
+};
+// 3 things it can do:
+//     1) Load a game.
+    void loadingGame(std::string gameApplicationToOpen);
+
+//     2) Connect peripherals.
+    int connectingDevice(std::string deviceName); // NOTE: returns a value to signify successful or failed connections.
 
+//     3) Display image.
+    void displayingToScreen(GamingLaptop gamingLaptopDisplay);
```
Comment (matkatmusic) [inline main.cpp]: These have not been declared as member functions
they are outside the struct's closing `}`
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Please review what a member function is and what makes a member function a member function and not a free function.
You have declared a bunch of free functions

File: main.cpp (hunk 3 above)
Comment (0bi0n3) [reply main.cpp]: Ah I was a bit confused with that, so thanks for clearing it up.

File: main.cpp (hunk 2 above)
Comment (0bi0n3) [reply main.cpp:978]: Thanks for this!

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**